doc/WebSite/Entity-History.md

```diff
@@ -113,9 +113,13 @@ While you can select tracked entities by configuration, you can use the
 ```
 
 All properties of MyEntity are tracked except MyProperty2 since it's
-explicitly disabled. The Audited attribute can be used to
+explicitly disabled. The **Audited** attribute can be used to
 save change logs for a desired property.
 
+When using **Audited** attribute on **entity** class, **EntityChange will always be created** irregardless of whether any **PropertyChange** is being created. 
+
+Similarly, when using **Audited** attribute on a **property**, **PropertyChange will always be created** irregardless of whether there is any difference between the **new and old values** of the **property**.
```

Comment:
This behaviour does not look right.
Unlike `EntityChange`, since the entity did change, in this case it is determined that the property did not change.

Reply:
This behavior was introduced in https://github.com/aspnetboilerplate/aspnetboilerplate/pull/4870
> mark property as to_be_removed if
> - property'a new value == old value
> - property does not have audited attribute
> - is not a shadow property

Reply:
i would think the bahavior of `Audited` attribute should be consistent when it is used on either `Entity` or `Property`, meaning changeset will always be created regardless if there is any changes.

If any user wants to create changeset only if `Entity` or `Property` has some changes, they should use `EntityHistory#Selectors`

Reply:
Currently, it is neither consistent nor correct.

- `EntityChange` is only created if `Entity` has changes<sup>1</sup>. (Correct)
- `PropertyChange` is created regardless of whether `Property` has changes. (Incorrect)

<sup>1</sup> regardless of whether those changes are not recorded in `PropertyChanges`, e.g. due to `DisableAuditing` as in #5093.

Reply:
> EntityChange is only created if Entity has  

This is not how it behaves at the moment. EntityChannge is always created for entity with Audited attribute

Reply:
Can you point me to the relevant code for that?

Because `CreateEntityChange` returns `null` in [EntityHistoryHelper.cs#L163-L182 as of v5.1.0](https://github.com/aspnetboilerplate/aspnetboilerplate/blob/v5.1.0/src/Abp.ZeroCore.EntityFrameworkCore/EntityHistory/EntityHistoryHelper.cs#L163-L182):
```c#
[CanBeNull]
private EntityChange CreateEntityChange(EntityEntry entityEntry)
{
    ...
    EntityChangeType changeType;
    switch (entityEntry.State)
    {
        case EntityState.Added:
            changeType = EntityChangeType.Created;
            break;
        case EntityState.Deleted:
            changeType = EntityChangeType.Deleted;
            break;
        case EntityState.Modified:
            changeType = entityEntry.IsDeleted() ? EntityChangeType.Deleted : EntityChangeType.Updated;
            break;
        case EntityState.Detached:
        case EntityState.Unchanged:
            return null;
```

Reply:
Ahh I think I didn’t elaborate my thoughts well enough.

The code your shown above is valid if Entity does not have any changes at all (no property has any changes). 

However , the scenario that I thought of was when there is a property with changes in the Entity but the property is marked with DisableAuditing attribute.

Reply:
That's exactly what I said:

> - `EntityChange` is only created if `Entity` has changes<sup>1</sup>. (Correct)
>
> <sup>1</sup> regardless of whether those changes are recorded in `PropertyChanges`, e.g. due to `DisableAuditing` as in #5093.